### ISSUE TITLE
Enable settings sync by default if localStorage is empty

### ISSUE THREAD

**Issue by xPaw** [thelounge/thelounge#2279]
Would be a good transition, it's also wanted now that logging out clears out localStorage.

**Comment by McInkay** [comment]
In reply to @astorije 

> I don't think I have mentioned this in the past (though I do have poor memory lol). 

Maybe I imagined it? I thought you had said it.

> One thing I'm wondering though: if we enable it by default, does it stay an "advanced setting"? I think so, but again, just clarifying the consequences/details.

It needs to be an advanced setting even more so when it's on by default.

> That I disagree with, however. We've discussed in the past that some settings should always be synced, because that's how they work on the server, such as custom highlights. Not having them synced simply means something is broken.

Yeah, but the problem is some settings always being synced is confusing. How do we make it clear to users that whether this setting is on or not, this one will be synced? My thinking of turning this on by default, we can say "Odd things might happen" if you disable it, around features like that.

> I actually don't think it will (or should) make it simpler to enable settings-sync by default than not. I think overall it is just 2 different options, for convenience/preference, but I don't foresee one or the other being easier to support.

Possibly, but I think it makes more sense, logically, to be enabled by default.

---

In reply to @creesch 

> This might be a bit more complicated than it seems on the surface. The first client that enables sync becomes the "master" and all settings for that client will be synced back to the server as the base set. Any client after that when sync is enabled will receive these settings.

Maybe we can solve this by allowing the user to set any future client overriding the server, rather than just taking the server settings by default.

> So as far as transition goes you might not want to do this in 3.0.0 where sync is introduced for the first time 

Actually, I think now is the best time to do it, because this is a short term issue. 

> A third option would be to implement a "force sync" sort of button that will sync all settings from that specific client as the new master going forward.

Oh wait, just saw you edited this into your comment. I'm good with this option. Combine that with enabling sync for all new clients, and soon everyone will default syncing.

**Comment by astorije** [comment]
> Yeah, but the problem is some settings always being synced is confusing. How do we make it clear to users that whether this setting is on or not, this one will be synced? My thinking of turning this on by default, we can say "Odd things might happen" if you disable it, around features like that.

I can think of a couple ways to improve the UI to make it clear that some parameter is always synced across devices, so I'm not concerned about that.
In a way, we already have some settings in the page that don't act like the other ones: resetting password and active sessions. I know it's different from what we are talking about, just pointing out that these exist, and we can live with it, and also build our UI to make this easy to work with.

I am however concerned to leave features half-broken when not syncing settings. This is just not something I want us to do. Imagine when bringing `autoAway` to the UI, do we completely disable the feature when not syncing? Keep it enabled but without guarantee it will work okay? That's not good imo.
And back to #1851, this was the whole point of `alwaysSync`: in a first pass, feed it highlights, and then whatever *needs* to be synced on the server to work correctly.

So in short, I completely get what you're saying, and yes there are UI and UX considerations to it, but we have room to make it good without sacrificing features that require server-sync, I think.

**Comment by McInkay** [comment]
> In a way, we already have some settings in the page that don't act like the other ones: resetting password and active sessions.

Yeah, I just couldn't think of any way of making it clear. However, if you can figure out a way to make it obvious, then that's fine.

Either way, this is separate to the functionality. I think, either way, that syncing should be the default. At the very least for new users, and I believe that it should be for new clients as well, and we can mitigate the issues that @creesch mentioned by allowing people to override the server settings when manually syncing a client that hasn't been synced before.

**Comment by astorije** [comment]
My understanding is that this was done in #2317 so closing. Feel free to re-open if I'm wrong.

**Comment by McInkay** [comment]
Correct, @astorije
